### ISSUE TITLE
Bad evaluation result on validation set [Solved]

### ISSUE THREAD

**Issue by Jokoe66** [HKUST-Aerial-Robotics/Stereo-RCNN#8]
I got this bad evaluation result on val set, as the following:
```
Thank you for participating in our evaluation!
Loading detections...
number of files for evaluation: 3446
  done.
save result//plot/car_detection.txt
car_detection AP: 18.119764 17.805336 17.665022
save result//plot/car_orientation.txt
car_orientation AP: 17.903425 17.490780 17.163260
save result//plot/car_detection_ground.txt
car_detection_ground AP: 11.219302 11.579988 9.090909
save result//plot/car_detection_3d.txt
car_detection_3d AP: 9.090909 9.090909 9.090909
Your evaluation results are available at:
result/
```
@PeiliangLi I use cuda8.0 and pytorch0.3.0 to run the model. The result ```.txt``` files are maintained below. ```trainlog.txt``` contains training log of the first 8 epochs. Is there any idea about this situation? Many thanks.
[result.zip](https://github.com/HKUST-Aerial-Robotics/Stereo-RCNN/files/3123411/result.zip)
[trainlog.txt](https://github.com/HKUST-Aerial-Robotics/Stereo-RCNN/files/3123418/trainlog.txt)

**Comment by PeiliangLi** [comment]
Did you write the result using our test_net.py? It seems that each of your result files only contains one object. That's the reason.

**Comment by Jokoe66** [comment]
> Did you write the result using our test_net.py? It seems that each of your result files only contains one object. That's the reason.

yes, I ran ```sh test.sh```

**Comment by Jokoe66** [comment]
> Did you write the result using our test_net.py? It seems that each of your result files only contains one object. That's the reason.

using your trained model, I can get relatively normal result.

**Comment by Jokoe66** [comment]
> Did you write the result using our test_net.py? It seems that each of your result files only contains one object. That's the reason.

Oh, sorry to see that I edited ```kitti_utils.write_detection_results``` and got the write mode as ```w``` instead of ```a```.

**Comment by PeiliangLi** [comment]
ok, please tell me if you get results using your trained model.

**Comment by Jokoe66** [comment]
> ok, please tell me if you get results using your trained model.

Yes, it's the reason. I can get results close to yours. Thanks.
